### PR TITLE
fix: Whitelist the "Authorization" header for CORS requests

### PR DESCRIPTION
**Pull request by kitcambridge**
Also add a "WWW-Authenticate" header for 401 responses, for maximum RFC 7235 compliance. :wink:

r=@bbangert

### CODE REVIEW

**Comment by bbangert** [comment]
I still see it as 'Bearer' in the vapid spec, not 'WebPush' for the Authorization header.

File: autopush/endpoint.py [hunk 1]
```diff
@@ -320,7 +325,7 @@ def _process_auth(self, result):
             except ValueError:
                 raise VapidAuthException("Invalid Authorization header")
             # if it's a bearer token containing what may be a JWT
-            if auth_type.lower() == 'bearer' and '.' in token:
+            if auth_type.lower() == AUTH_SCHEME and '.' in token:
```
Comment (jrconlin) [inline autopush/endpoint.py]: We probably want to make this 
   if auth_type.lower() in [AUTH_SCHEME, 'bearer'] and ...
for a few releases. There are a few folks in the wild using 'bearer' and we don't want to break them yet.

File: autopush/endpoint.py [hunk 2]
```diff
@@ -768,7 +774,7 @@ def _validate_auth(self, uaid):
                                         header.strip()).split(" ", 2)
         except ValueError:
             return False
-        if "bearer" != token_type.lower():
+        if AUTH_SCHEME != token_type.lower():
```
Comment (jrconlin) [inline autopush/endpoint.py]: See above.

**Comment by bbangert** [comment]
Just missing a changelog.... :grin:

**Pull request closed without merging**